File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def isPalindrome(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: bool
        """
        arr=[]
        temp=head
        while temp:
            arr.append(temp.val)
            temp=temp.next
        return arr==list(reversed(arr))
        
        

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
     List<Integer> list=new ArrayList<>();
     ListNode temp=head;
     while(temp!=null){
        list.add(temp.val);
        temp=temp.next;
     }   
     temp=head;

     for(int i=list.size()-1;i>=0;i--){
        if(list.get(i)!=temp.val) return false;
        temp=temp.next;
     }
     return true;
    }
}